spec/jobs/connect_app_job_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe ConnectAppJob, type: :job, vcr: true do
+  subject { ConnectAppJob.new(app, token) }
+
+  let(:app) { ConnectedApp.create!(enterprise: ) }
+  let(:enterprise) { create(:enterprise) }
+  let(:token) { enterprise.owner.spree_api_key }
+
+  before { enterprise.owner.generate_api_key }
+
+  it "stores connection data on the app" do
+    subject.perform_now
+
+    expect(app.data).to eq({ "link" => "https://example.net/edit" })
```

Comment:
The URL will change on the live endpoint once it's implemented. We may not be able to do live recordings then because we don't want to add our test enterprise to the live database. Maybe we can have a test mode, let's see.